src/ast/variables/Variable.ts

```diff
@@ -1,16 +1,15 @@
-/* eslint-disable no-unused-vars */
-
-import { UNKNOWN_ASSIGNMENT, PredicateFunction } from '../values';
+import { UNKNOWN_EXPRESSION, UNKNOWN_VALUE } from '../values';
 import CallOptions from '../CallOptions';
 import ExecutionPathOptions from '../ExecutionPathOptions';
 import Identifier from '../nodes/Identifier';
-import { ForEachReturnExpressionCallback } from '../Node';
 import { ObjectPath } from './VariableReassignmentTracker';
+import { Expression, ForEachReturnExpressionCallback, SomeReturnExpressionCallback } from '../nodes/shared/Expression';
 
-export default class Variable {
+export default class Variable implements Expression {
```

Comment:
How about ExpressionEntity as the base shared between Variable and Node?

Then we've have `ExpressionEntity`, `ExpressionNode` and `ExpressionBase`?

Reply:
I like that. So that's what it's gonna be!